lib/providers/filesystem/index.js

```diff
@@ -40,9 +40,10 @@ function FileSystemProvider(options) {
 }
 
 var namePattern = new RegExp('[^' + path.sep + '/]+');
+var badPath = /(^|[\\\/])\.\.([\\\/]|$)/;
```

Comment:
IMO, security code should be clear on what it is doing. A comment would be good. Its looking for any of
`..`, `../blah`, `blah/..`, or `blah/../blah`, right? A better name would be `containsDotDotPaths()`.